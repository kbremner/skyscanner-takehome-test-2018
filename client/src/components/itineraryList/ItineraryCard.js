import React from 'react';
import BpkPanel from 'bpk-component-panel';
import PropTypes from '../../types';
import ItineraryLeg from './ItineraryLeg';
import ItineraryFooter from './ItineraryFooter';
import './ItineraryCard.scss';

const ItineraryCard = ({ className, itinerary }) => (
  <BpkPanel className={[className, "itinerary-card"].join(' ')}>
    {itinerary.legs.map(({ id, carrier, duration, numStops, segments }) => (
      <ItineraryLeg
        key={id}
        className="itinerary-card__leg"
        carrier={carrier}
        duration={duration}
        numStops={numStops}
        segments={segments}
      />
    ))}
    <ItineraryFooter
      className="itinerary-card__footer"
      price={itinerary.price}
      agentName={itinerary.agentName}
      bookingLink={itinerary.bookingLink}
    />
  </BpkPanel>
);

ItineraryCard.propTypes = {
  itinerary: PropTypes.itinerary.isRequired,
  className: PropTypes.string
};

ItineraryCard.defaultProps = {
  className: null
};

export default ItineraryCard;