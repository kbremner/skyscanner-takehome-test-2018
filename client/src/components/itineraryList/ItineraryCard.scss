@import "~bpk-mixins";

.itinerary-card {
  box-shadow: $bpk-box-shadow-sm;
}