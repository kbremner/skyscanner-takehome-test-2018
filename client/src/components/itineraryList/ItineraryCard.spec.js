import React from 'react';
import { shallow } from 'enzyme';
import ItineraryCard from './ItineraryCard';

const defaultItinerary = (id) => ({
  price: '$123.45',
  agentName: `test agent`,
  bookingLink: `http://example.com/${id}`,
  legs: [
    {
      id: '1',
      carrier: {
        name: 'easyjet',
        code: 'EZ'
      },
      duration: 210,
      numStops: 0,
      segments: [11, 12, 13, 14].map(val => ({
        origin: {
          dateTime: `${val}:00`, code: `CODE-${id}-${val}`
        },
        destination: {
          dateTime: `${val+1}:00`, code: `CODE-${id}-${val+1}`
        }
      }))
    }
  ]
});

const setup = (itineraryOverrides = {}, className = null) => {
  const itinerary = {
    ...defaultItinerary('1'),
    ...itineraryOverrides
  };

  const onClick = jest.fn();

  const wrapper = shallow(<ItineraryCard
    className={className}
    itinerary={itinerary}
    onClick={onClick}
  />);

  return { wrapper, itinerary, className, onClick };
};

describe('ItineraryCard', () => {
  it('renders', () => {
    const { wrapper } = setup();
    expect(wrapper).toMatchSnapshot();
  });

  it('adds class name to card provided', () => {
    const { wrapper, className } = setup({}, 'test-class-name');
    expect(wrapper.hasClass(className)).toBe(true);
  });
});
