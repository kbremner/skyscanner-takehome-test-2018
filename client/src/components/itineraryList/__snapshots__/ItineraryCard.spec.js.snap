// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItineraryCard renders 1`] = `
<BpkPanel
  className=" itinerary-card"
  fullWidth={false}
  padded={true}
>
  <ItineraryLeg
    carrier={
      Object {
        "code": "EZ",
        "name": "easyjet",
      }
    }
    className="itinerary-card__leg"
    duration={210}
    key="1"
    numStops={0}
    segments={
      Array [
        Object {
          "destination": Object {
            "code": "CODE-1-12",
            "dateTime": "12:00",
          },
          "origin": Object {
            "code": "CODE-1-11",
            "dateTime": "11:00",
          },
        },
        Object {
          "destination": Object {
            "code": "CODE-1-13",
            "dateTime": "13:00",
          },
          "origin": Object {
            "code": "CODE-1-12",
            "dateTime": "12:00",
          },
        },
        Object {
          "destination": Object {
            "code": "CODE-1-14",
            "dateTime": "14:00",
          },
          "origin": Object {
            "code": "CODE-1-13",
            "dateTime": "13:00",
          },
        },
        Object {
          "destination": Object {
            "code": "CODE-1-15",
            "dateTime": "15:00",
          },
          "origin": Object {
            "code": "CODE-1-14",
            "dateTime": "14:00",
          },
        },
      ]
    }
  />
  <ItineraryFooter
    agentName="test agent"
    bookingLink="http://example.com/1"
    className="itinerary-card__footer"
    price="$123.45"
  />
</BpkPanel>
`;
